### PR TITLE
Update manager suspend and settings exec endpoints

### PR DESCRIPTION
**Pull request by adairrr**
This change does the following:
- Update status and setting endpoints in the manager
- Add `AbstractAttributes` trait and tests

`EnableIBC` -> `UpdateSettings`
`SuspendAccount` -> `UpdateStatus`

The `AbstractAttributes` trait allows users to add abstract attributes directly to their response.

### CODE REVIEW

File: contracts/account/manager/src/commands.rs [hunk 1]
```diff
@@ -398,7 +401,7 @@ pub fn replace_api(
         old_api_addr.into_string(),
     )?);
     // Add new api to proxy
-    msgs.push(whitelist_dapp_on_proxy(
+    msgs.push(allowlist_dapp_on_proxy(
```
Comment (CyberHoward) [inline contracts/account/manager/src/commands.rs:404]: haha bruh this

File: contracts/account/manager/src/commands.rs [hunk 2]
```diff
@@ -430,34 +433,40 @@ pub fn update_info(
...
 
     // Only the subscription contract can load
     if let Some(sub_addr) = config.subscription_address {
         if sub_addr.eq(&info.sender) {
-            STATUS.save(deps.storage, &new_status)?;
-            return Ok(ManagerResponse::new(
-                "update_subscription_status",
-                vec![("new_status", new_status.to_string())],
-            ));
+            SUBSCRIPTION_STATUS.save(deps.storage, &suspension_status)?;
```
Comment (CyberHoward) [inline contracts/account/manager/src/commands.rs:447]: so i'm still not convinced that we should have this be related to "subscriptions". Can we just store it as SUSPENTION_STATUS?

File: contracts/account/manager/src/commands.rs [hunk 3]
```diff
@@ -430,34 +433,40 @@ pub fn update_info(
...
     deps: DepsMut,
     info: MessageInfo,
-    new_status: Subscribed,
+    suspension_status: Subscribed,
+    response: Response,
 ) -> ManagerResult {
     let config = CONFIG.load(deps.storage)?;
 
     // Only the subscription contract can load
     if let Some(sub_addr) = config.subscription_address {
```
Comment (CyberHoward) [inline contracts/account/manager/src/commands.rs:445]: so same here, i'm thinking of removing the subscription address from the contract's state.

File: packages/abstract-core/src/core/manager.rs [hunk 4]
```diff
@@ -134,13 +126,11 @@ pub enum ExecuteMsg {
...
-    SuspendAccount {
-        new_status: bool,
-    },
-    EnableIBC {
-        new_status: bool,
-    },
+    /// Update suspension status
+    UpdateStatus { suspend: Option<bool> },
+    /// Update settings for the Account, including IBC enabled, etc.
+    UpdateSettings { enable_ibc: Option<bool> },
```
[the remark this answers is not in the document]
Comment (CyberHoward) [reply packages/abstract-core/src/core/manager.rs:132]: sure!

**Comment by adairrr** [comment]
Task linked: [CU-861mkce4z Remove subscription-related items from factory and manager](https://app.clickup.com/t/861mkce4z)

**Comment by CyberHoward** [comment]
update on above comments?

**Comment by adairrr** [comment]
> update on above comments?
Fixed and waiting for CI

**Review by CyberHoward — APPROVED** [review]

**Comment by adairrr** [comment]
Task linked: [CU-861mkcfhf Fix misleading message names](https://app.clickup.com/t/861mkcfhf)

**Pull request merged**